services/121-service/src/events/entities/event.entity.ts

```diff
@@ -16,7 +16,7 @@ import { UserEntity } from '@121-service/src/user/user.entity';
 
 @Entity('event')
 export class EventEntity extends Base121OptionalAuditedEntity {
-  @ManyToOne(() => UserEntity)
+  @ManyToOne(() => UserEntity, { onDelete: 'CASCADE' })
```

Comment:
I am wondering a bit about this. If you delete a user would you also want to delete all events related to that user (so the audit trail of updating registrations, but also the transactions created by that user). I also found this in the user service at the delete endpoint:

```
    try {
      return await this.userRepository.remove(user);
    } catch (e) {
      if (e.code === '23503') {
        throw new HttpException(
          'User cannot be removed because it is related to other entities for logging purposes',
          HttpStatus.CONFLICT,
        );
      }
      throw e;
    }
```
So before we sort of consciously made a decision no to delete users? 

Maybe we should pick one of these options for users:
1. Make userId optional in transactions and events entities
2. Set the `active` boolean of users to false instead of deleting them. And maybe allowing updates to the username (so you can anomize the email adress)
3. Give extra warnings when deleting users with transactions?

Reply:
yeah good point. It was still the goal of [this to do task](https://dev.azure.com/redcrossnl/121%20Platform/_sprints/taskboard/121%20Development%20Team/121%20Platform/Sprint%20144?workitem=31938) to go over relationships in detail again and notice things like this. 

Let's discuss indeed.